nixos/modules/services/desktops/gnome3/gnome-initial-setup.nix

```diff
@@ -4,6 +4,44 @@
 
 with lib;
 
+let
+
+  # GNOME initial setup's run is conditioned on whether
+  # the gnome-initial-setup-done file exists in XDG_CONFIG_HOME
+  # Because of this, every existing user will have initial setup
+  # running because they never ran it before.
+  #
+  # To prevent this we create the file if the users stateVersion
+  # is older than 20.03 (the release we added this module).
```

Comment:
Indeed this is terrifying :smile: 

No idea why this wasn't rewritten to use gsettings, Pantheon's onboarding application keeps a key of pages viewed to condition it starting. With something like that this could become a simple gsettings override.